### PR TITLE
[SCB-2643]fix instance isolation filter not work problem

### PR DESCRIPTION
**Pull request by zhaozw**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install -Pit` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: handlers/handler-loadbalance/src/main/java/org/apache/servicecomb/loadbalance/ServerListFilterExt.java [hunk 1]
```diff
@@ -29,7 +29,7 @@
  */
 public interface ServerListFilterExt {
   default int getOrder() {
-    return 0;
+    return FilterConstant.ORDER_NORMAL;
```
Comment (liubao68) [inline handlers/handler-loadbalance/src/main/java/org/apache/servicecomb/loadbalance/ServerListFilterExt.java:32]: Do not change default value. ORDER_NORMAL should be 0.

File: handlers/handler-loadbalance/src/main/java/org/apache/servicecomb/loadbalance/FilterConstant.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.servicecomb.loadbalance;
+
+public interface FilterConstant {
+  int ORDER_NORMAL = 100;
```
Comment (liubao68) [inline handlers/handler-loadbalance/src/main/java/org/apache/servicecomb/loadbalance/FilterConstant.java:21]: 0

File: handlers/handler-loadbalance/src/main/java/org/apache/servicecomb/loadbalance/FilterConstant.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  int ORDER_ISOLATION = 0;
```
Comment (liubao68) [inline handlers/handler-loadbalance/src/main/java/org/apache/servicecomb/loadbalance/FilterConstant.java:23]: -100

File: handlers/handler-loadbalance/src/main/java/org/apache/servicecomb/loadbalance/FilterConstant.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  int ORDER_ZONE_AWARE = 200;
+
+  String EMPTY_INSTANCE_PROTECTION = "servicecomb.loadbalance.filter.isolation.emptyInstanceProtectionEnabled";
```
Comment (liubao68) [inline handlers/handler-loadbalance/src/main/java/org/apache/servicecomb/loadbalance/FilterConstant.java:27]: Maybe can define these constants in ServerListFilterExt

File: handlers/handler-loadbalance/src/main/java/org/apache/servicecomb/loadbalance/filterext/IsolationDiscoveryFilter.java [hunk 5]
```diff
@@ -42,17 +39,15 @@
...
+public class IsolationDiscoveryFilter implements ServerListFilterExt {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(IsolationDiscoveryFilter.class);
 
-  private static final String EMPTY_INSTANCE_PROTECTION = "servicecomb.loadbalance.filter.isolation.emptyInstanceProtectionEnabled";
-
   private final DynamicBooleanProperty emptyProtection = DynamicPropertyFactory.getInstance()
       .getBooleanProperty(EMPTY_INSTANCE_PROTECTION, false);
 
+  public EventBus eventBus = EventManager.getEventBus();
```
Comment (liubao68) [inline handlers/handler-loadbalance/src/main/java/org/apache/servicecomb/loadbalance/filterext/IsolationDiscoveryFilter.java:49]: use private

Comment (Shoothzj) [reply handlers/handler-loadbalance/src/main/java/org/apache/servicecomb/loadbalance/filterext/IsolationDiscoveryFilter.java]: can we mark this field final?

**Review by liubao68 — APPROVED** [review]

**Comment by etteeee** [comment]
Now we have changed isolation and zoneaware type from `DiscoveryFilters` to `ServerListFilterEx`, so what is the real difference between this 2 types filter except for the filtering timing? What is the **RULE** when adding a new filter policy? @liubao68 @zhaozw 

btw, it's better to create a new class with new name rather than moving to a new package, the class name may cause unnecessary confusion.

**Comment by liubao68** [comment]
@etteeee DiscoveryFilters may have better performance but can only be used when filter results not change only when instance lists changed, while ServerListFilterEx can be used for general purpose.

**Pull request merged**